tt_eager/tt_dnn/op_library/transpose/wh_multi_core/transpose_wh_op_multi_core.cpp

```diff
@@ -362,16 +362,16 @@ operation::ProgramWithCallbacks transpose_wh_multi_core_sharded(const Tensor &a,
         const std::vector<Tensor>& output_tensors
     ) {
 
-        const auto& src_tensor = input_tensors.at(0);
-        const auto& dst_tensor = output_tensors.at(0);
+        const auto src_tensor = input_tensors.at(0);
+        const auto dst_tensor = output_tensors.at(0);
```

Comment:
Why do we remove ref here?

Reply:
I think it was unnecessary here tbh, but below where we get the shard_spec from ref and use it in bounding boxes, I think the variable it was referring to goes out of bounds and I end up getting a set fault at the `all_cores.bounding_box` line.

Reply:
I can set the ref back for this and just leave it for shard_spec